### PR TITLE
Giftcard fixes

### PR DESCRIPTION
**Pull request by abbasali1994**
Ticket(s):
- [1.7.1 testing - gift cards#1833](https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1833)

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
- go to http://localhost:4200/gift

## UI changes for review

image
image
image
image

### CODE REVIEW

File: src/pages/Gift/Purchase/Result/Success.tsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ function EmailCode({ secret }: { secret: string }) {
         </p>
       </div>
       <Link
-        to={`../${routes.mail}`}
+        to={`${routes.mail}`}
```
Comment (misicnenad) [inline src/pages/Gift/Purchase/Result/Success.tsx:71]: nit: ``to={`${routes.mail}`}`` can be changed to `to={routes.mail}`

File: src/pages/Gift/Mailer/Form.tsx [hunk 2]
```diff
@@ -15,11 +16,18 @@ export default function Form({ classes = "" }) {
   const { handleSubmit, reset } = useFormContext<FV>();
```
Comment (ap-justin) [inline src/pages/Gift/Mailer/Form.tsx:16]:
```suggestion
  const { handleSubmit, reset, formState:{ isSubmitting } } = useFormContext<FV>();
```
you could use this state unless there's a reason for separate `isSending` state ?

**Review by ap-justin — CHANGES_REQUESTED** [review]

**Review by misicnenad — CHANGES_REQUESTED** [review]
Is the issue with the tx estimation error solved now (see [Tim's comment](https://app.clickup.com/t/8677d7yzw?comment=90100005598414))?

Also, one of the requirements from the attached Loom video (see [CU task description](https://app.clickup.com/t/8677d7yzw)) was that after the "Send Giftcard" submit btn click, the user gets navigated away from the page - we should confirm which page that should be.

**Comment by abbasali1994** [comment]
Waiting for Tim to provide confirmation on which page to redirect after closing prompt

**Comment by SovereignAndrey** [comment]
> Waiting for Tim to provide confirmation on which page to redirect after closing prompt

I believe he commented that the Marketplace page would be the redirect destination.

**Comment by misicnenad** [comment]
@SovereignAndrey @abbasali1994 @ap-justin created a new "Successful sending of gift code" component in accordance with the new requirements ([link to Lambert's comment](https://app.clickup.com/t/8677d7yzw?comment=90050006800184&threadedComment=90030010700631)).

The email format request should be handled on AWS side @jp-mariano

**Review by ap-justin — APPROVED** [review]

**Pull request merged**